crates/sui-core/src/state_accumulator/mod.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use sui_types::committee::EpochId;
+use typed_store::Map;
+
+use std::sync::Arc;
+use sui_types::base_types::ObjectDigest;
+
+use fastcrypto::hash::{Digest, MultisetHash};
+use sui_types::accumulator::Accumulator;
+use sui_types::error::SuiResult;
+use sui_types::messages::TransactionEffects;
+use sui_types::messages_checkpoint::CheckpointSequenceNumber;
+use typed_store::rocks::TypedStoreError;
+
+use crate::authority::authority_per_epoch_store::AuthorityPerEpochStore;
+use crate::authority::AuthorityStore;
+
+#[derive(Debug, Clone)]
+pub struct State {
```

Comment:
I think this should probably be called `StateAccumulationRequest` or something. Also, maybe we don't need this struct anymore? functions can just take `effects` and `checkpoint_seq_num` as arguments?

Reply:
Yeah agreed, can just remove